src/redux/actions/experimentSettings/processingConfig/updateFilterSettings.js

```diff
@@ -3,6 +3,17 @@ import { EXPERIMENT_SETTINGS_SAMPLE_FILTER_UPDATE, EXPERIMENT_SETTINGS_NON_SAMPL
 const settingsWithSampleId = ['classifier', 'cellSizeDistribution', 'doubletScores', 'mitochondrialContent', 'numGenesVsNumUmis'];
 const settingsWithoutSampleId = ['dataIntegration', 'configureEmbedding', 'meta', 'defaultFilterSettings'];
 
+/**
+ * Updates filter settings for a step in data processing
+ *
+ * @param {*} step The key of the step that is changing.
+ * @param {*} diff The change itself that is taking place.
+ * @param {*} sampleId The id of the sample that had its settings changed
+ * (if the settings of the step are sample specific).
+ * @param {*} isALocalChange Whether this is an update received from qc running
```

Comment:
Why do we need to know this? Just interested.

Reply:
I just felt like it was useful to add comments for this since it has a lot of important optional parameters that change what the action creator does based on whether we pass them or not

Reply:
i meant what is the variable `isALocalChange` used for sorry :D

Reply:
oh, I just cleaned it up a littel bit because it isn't used as much, you can check the reducer for EXPERIMENT_SETTINGS_NON_SAMPLE_FILTER_UPDATE, we use it to know whether we should update `originalProcessing` or not.

We use `originalProcessing` to keep the version of processingConfig that we want to go back to when a user clicks "Discard".

We only need to update `originalProcessing` when the change is not local, so when we received an update from qc saying that this is the representative value for the displayed information as opposed to local changes which are just the current user clicking on buttons without clicking run yet).